Include/cpython/listobject.h

```diff
@@ -21,6 +21,13 @@ typedef struct {
     Py_ssize_t allocated;
 } PyListObject;
 
+#ifdef Py_GIL_DISABLED
+typedef struct {
+    Py_ssize_t allocated;
+    PyObject *ob_item[1];
+} _PyListArray;
+#endif
```

Comment:
Let's put this in `listobject.c` (or `pycore_list.h` if it needs to be in a header).